### PR TITLE
Semantic versioning for osmosis v4.x

### PR DESCRIPTION
**Pull request by faddat**
## Description

Made Osmosis be able to do versioned imports, go-style.   Used the "mod" tool below. 

﻿github.com/marwan-at-work/mod

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ValarDragon** [comment]
Oh sweet, glad to know that theres tooling to help with this! Looks like it missed a couple items, once tests/build passes, good to merge imo.

Thanks for doing this!

**Comment by faddat** [comment]
nice, I'll check out the issues now.  This was a blind-submit.

**Comment by faddat** [comment]
This is failing on linters; my plan is to fix e'm so that we've a nice, clean v4.2.0 to reference in the future.

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/golang/protobuf v1.5.2
 	github.com/gorilla/mux v1.8.0
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
+	github.com/osmosis-labs/osmosis v1.0.4
```
Comment (ValarDragon) [inline go.mod:13]: We missed an import somewhere right? I don't think we should be importing v1 anywhere

Comment (faddat) [reply go.mod:13]: definitely.  I should have marked this one as draft.  

v3 is ready to rock.

**Pull request closed without merging**